plugin-default-impl/src/main/java/com/alibaba/nacos/plugin/auth/impl/token/impl/CachedJwtTokenManager.java

```diff
@@ -0,0 +1,245 @@
+/*
+ * Copyright 1999-2021 Alibaba Group Holding Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.alibaba.nacos.plugin.auth.impl.token.impl;
+
+import com.alibaba.nacos.plugin.auth.exception.AccessException;
+import com.alibaba.nacos.plugin.auth.impl.token.TokenManager;
+import com.alibaba.nacos.plugin.auth.impl.users.NacosUser;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.scheduling.annotation.Scheduled;
+import org.springframework.security.core.Authentication;
+import org.springframework.stereotype.Component;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Cached JWT token manager.
+ *
+ * @author majorhe
+ */
+@Component
+public class CachedJwtTokenManager implements TokenManager {
+    
+    //key: token string, value: token entity
+    private volatile Map<String, TokenEntity> tokenMap = new ConcurrentHashMap<>(1024);
+    
+    //key: username, value: token entity. cache token created by self.
+    private volatile Map<String, TokenEntity> userMap = new ConcurrentHashMap<>(128);
```

Comment:
`username` and `TokenEntity` is a one-to-one relationship?
I think the same user may have multiple tokens at the same time.

Reply:
yes, it aims to reduce token quantity